packages/kakarot/src/kakarot.ts

```diff
@@ -0,0 +1,528 @@
+// Credits to https://github.com/wevm/wagmi for the inspiration on the implementation
+// for the kakarot connector
+
+import type {
+  AddInvokeTransactionParameters,
+  Call as RequestCall,
+  RequestFnCall,
+  RpcMessage,
+  RpcTypeToMessageMap,
+  SwitchStarknetChainParameters,
+  TypedData,
+} from "@starknet-io/types-js";
+import { mainnet, sepolia } from "@starknet-react/chains";
+import {
+  type ChainProviderFactory,
+  Connector,
+  ConnectorNotFoundError,
+} from "@starknet-react/core";
+import type {
+  ConnectorData,
+  ConnectorIcons,
+} from "@starknet-react/core/src/connectors/base";
+import type { EIP6963ProviderDetail } from "mipd";
+import {
+  Account,
+  type AccountInterface,
+  type ProviderInterface,
+  type ProviderOptions,
+  type RpcProvider,
+  hash,
+} from "starknet";
+import {
+  type EIP1193Provider,
+  type ProviderConnectInfo,
+  type ProviderMessage,
+  type RpcError,
+  type WalletClient,
+  encodeAbiParameters,
+  getAddress,
+  numberToHex,
+  toHex,
+  withTimeout,
+} from "viem";
+import {
+  KAKAROT_DEPLOYMENTS,
+  getCorrespondingKakarotChain,
+  getCorrespondingStarknetChain,
+} from "./chains";
+
+const MULTICALL_CAIRO_PRECOMPILE = "0x0000000000000000000000000000000000750003";
+
+class ProviderNotFoundError extends Error {
+  constructor() {
+    super("Provider not found.");
+  }
+}
+
+class ChainIdInvalidError extends Error {
+  constructor(chainId: number) {
+    super(`Chain id invalid: ${chainId}`);
+  }
+}
+
+type EthereumConnectorEvents = {
+  onAccountsChanged(accounts: string[]): void;
+  onChainChanged(chainId: string): void;
+  onConnect?(connectInfo: ProviderConnectInfo): void;
+  onDisconnect(error?: Error | undefined): void;
+  onMessage?(message: ProviderMessage): void;
+};
+
+let accountsChanged: EthereumConnectorEvents["onAccountsChanged"] | undefined;
+let chainChanged: EthereumConnectorEvents["onChainChanged"] | undefined;
+let connect: EthereumConnectorEvents["onConnect"] | undefined;
+let disconnect: EthereumConnectorEvents["onDisconnect"] | undefined;
+
+export interface KakarotConnectorOptions {
+  id: string;
+  name: string;
+  icon: ConnectorIcons;
+}
+export class KakarotConnector extends Connector {
```

Comment:
as I understand, some methods comply with starknet wallet spec, via starknet react, other methods comply with Eth EIP1193

Should we add in docstring methods linked with Starknet and others for Eth?